web/src/main/java/no/nav/ung/sak/web/app/ungdomsytelse/UtbetalingstatusUtleder.java

```diff
@@ -0,0 +1,110 @@
+        var sortert = tidslinjeMap.entrySet().stream()
+            .sorted(Comparator.comparing(e -> e.getKey().avsluttetTid(), Comparator.nullsLast(Comparator.naturalOrder())))
+            .toList();
+        return sortert;
+    }
+
+    private static LocalDateTimeline<UtbetalingStatus> finnUtbetalingstatusForEndring(LocalDate dagensDato, LocalDateTimeline<Boolean> endretTidslinje, BehandlingAvsluttetTidspunkt behandlingAvsluttetTidspunkt) {
+        var sisteUtbetalteDag = finnSisteMuligeUtbetalteDag(dagensDato, behandlingAvsluttetTidspunkt);
+        var statusForEndretYtelse = endretTidslinje.map(it -> it.getTom().isAfter(sisteUtbetalteDag) ?
+            List.of(new LocalDateSegment<>(it.getFom(), it.getTom(), UtbetalingStatus.TIL_UTBETALING)) :
+            List.of(new LocalDateSegment<>(it.getFom(), it.getTom(), UtbetalingStatus.UTBETALT)));
+        return statusForEndretYtelse;
+    }
+
+    private static LocalDate finnSisteMuligeUtbetalteDag(LocalDate dagensDato, BehandlingAvsluttetTidspunkt behandlingAvsluttetTidspunkt) {
+        var førsteVirkedagINåværendeMåned = finnFørsteVirkedagIMåned(YearMonth.from(dagensDato));
+
+
+        var sisteMuligeDatoForOversendingTilOS = finnDagForOversendingAvForrigeMåned(behandlingAvsluttetTidspunkt, førsteVirkedagINåværendeMåned, dagensDato);
+
+        var sisteMuligeDagForUtbetalingTilBruker = sisteMuligeDatoForOversendingTilOS.plusDays(3);
+        var sisteUtbetalteDag = !dagensDato.isBefore(sisteMuligeDagForUtbetalingTilBruker) ?
+            dagensDato.minusMonths(1).with(TemporalAdjusters.lastDayOfMonth()) : dagensDato.minusMonths(2).with(TemporalAdjusters.lastDayOfMonth());
+        return sisteUtbetalteDag;
+    }
+
+    private static LocalDate finnDagForOversendingAvForrigeMåned(BehandlingAvsluttetTidspunkt behandlingAvsluttetTidspunkt, LocalDate førsteVirkedagINåværendeMåned, LocalDate dagensDato) {
+        if (!behandlingAvsluttetTidspunkt.erAvsluttet()) {
+            // Hvis avsluttetDato er null, så betyr det at behandlingen ikke er avsluttet og vi antar at oversendingen skjer i dag
+            return dagensDato;
+        }
+        return behandlingAvsluttetTidspunkt.avsluttetTid().toLocalDate().isAfter(førsteVirkedagINåværendeMåned) ? behandlingAvsluttetTidspunkt.getAvsluttetTid().toLocalDate() : førsteVirkedagINåværendeMåned;
+    }
+
+    private static LocalDateTimeline<Boolean> finnEndretFraForrigeTidslinje(LocalDateTimeline<TilkjentYtelseVerdi> gjeldendeTilkjentYtelse, LocalDateTimeline<TilkjentYtelseVerdi> tilkjentYtelseTidslinje) {
+        return tilkjentYtelseTidslinje.combine(gjeldendeTilkjentYtelse, (di, lhs, rhs) -> {
+            if (rhs == null) {
+                return new LocalDateSegment<>(di, true);
+            }
+            if (!lhs.getValue().equals(rhs.getValue())) {
```

Comment:
Her sjekkes ikke lhs for null.

Foreslår å bruke noe sånt:
```java
if (!Objects.equals(lhs == null ? null : lhs.getValue(), rhs == null ? null : rhs.getValue())) {
    return new LocalDateSegment<>(di, true);
}
```

Reply:
Det er left-join og då er ikkje nullsjekk av lhs nødvendig

Reply:
Combine tar inn 3 parametere; ein tidlinje, ein kombinator og ein join style